birdhouse/scripts/get-components-json.include.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+# Obtain a JSON representation of components enabled on this platform.
+#
+# Expected result should be similar to:
+# {
+#  "components": [
+#    "bird-house/birdhouse-deploy:components/monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/canarie-api-full-monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/all-public-access",
+#    "bird-house/birdhouse-deploy:optional-components/wps-healthchecks",
+#    "bird-house/birdhouse-deploy:optional-components/secure-thredds",
+#    "bird-house/birdhouse-deploy:optional-components/testthredds",
+#    "bird-house/birdhouse-deploy:optional-components/test-weaver",
+#    "bird-house/birdhouse-deploy:components/weaver",
+#    "bird-house/birdhouse-deploy:components/cowbird"
+#  ]
+# }
+#
+
+# default value in case of error or missing definitions
+export BIRDHOUSE_DEPLOY_COMPONENTS_JSON='{"components": []}'
+if [ -z "${EXTRA_CONF_DIRS}" ]; then
+  echo "No components in EXTRA_CONF_DIRS. Components JSON list will be empty!"
+  exit 0
+fi
+
+# resolve path considering if sourced or executed, and whether from current dir, pavics-compose include or another dir
+BIRDHOUSE_DEPLOY_COMPONENTS_ROOT=$(dirname -- "$(realpath "$0")")
+if [ "$(echo "${BIRDHOUSE_DEPLOY_COMPONENTS_ROOT}" | grep -cE "/birdhouse/?\$" 2>/dev/null)" -eq 1 ]; then
+  BIRDHOUSE_DEPLOY_COMPONENTS_ROOT=.
+else
+  BIRDHOUSE_DEPLOY_COMPONENTS_ROOT="${BIRDHOUSE_DEPLOY_COMPONENTS_ROOT}/.."
+fi
+chdir "${BIRDHOUSE_DEPLOY_COMPONENTS_ROOT}"
+
+# note: no quotes in 'ls' on purpose to expand glob patterns
+BIRDHOUSE_DEPLOY_COMPONENTS_LIST_KNOWN="$( \
+  ls -d1 ./*components/*/ \
+  | sed -E "s|\./(.*)/|\1|" \
+)"
+if [ -z "${BIRDHOUSE_DEPLOY_COMPONENTS_LIST_KNOWN}" ]; then
+  echo "[ERROR]" \
+    "Could not resolve known birdhouse-deploy components." \
+    "Aborting to avoid potentially leaking sensible details."
+  exit 1
```

Comment:
I'd rather just put a warning here and do not `exit 1` to avoid leaving the entire platform down during unattended autodeploy.

Reply:
Ok, but in that case I will enforce the components to be empty as failing to retrieve those will not let us perform the filtering.